### ISSUE TITLE
Node gouping and sub-graphs

### ISSUE THREAD

**Issue by leissler** [McManning/BlueGraph#53]
In the dialog example, it seems that there is a grouping feature for nodes. However, when moving the group, it turns out that it's just an empty node around a couple of nodes and doesn't really group them.

So I think that a real grouping feature that treats all nodes in a group as one entity would be very useful since larger graphs quickly become hard to manage. In the best case, such a group should act like a sub-graph with its own inputs and outputs that can be re-used like a function.

**Comment by McManning** [comment]
Originally I had written it to behave similar to UE4 blueprints, where nodes are moved with the container comment as you suggested. But I was running into a good amount of undesirable edge cases that made me strip it from the 1.0 release. I've got #5 open already for this one.

For subgraphs, ideally I'd like to keep that as a separate feature where you can extract a group of nodes into a separate graph asset with explicit inputs and outputs - similar to [UE4 macros](https://docs.unrealengine.com/4.27/en-US/ProgrammingAndScripting/Blueprints/BP_HowTo/MacroLibraries/). I know I started on an implementation but it never made it into the samples project. My graphs tend to be pretty lightweight so I haven't spent much time on quality of life features for large graphs.

**Comment by leissler** [comment]
I agree that comment-move and sub-graphs are two separate issues. So are you planning on integrating real subgraphs similar to UE4 macros?
